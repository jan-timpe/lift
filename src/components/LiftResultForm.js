import React, { Component } from 'react'
import '../style/Form.css'


export default class LiftResultForm extends Component {

  constructor(props) {
    super(props)
    this.state = {
      weight: ''
    }
  }

  render() {
    return (
      <form onSubmit={ (event) => {
        event.preventDefault()
        let date = new Date()
        this.props.addLiftResult({
          liftId: this.props.liftId,
          weight: parseInt(this.state.weight),
          date: date.toLocaleDateString('en-US')
        })
        this.setState({ weight: '' })
      }}>

        <div className="input-wrapper">
          <label htmlFor="weight">Add result:</label>
          <input type="number"placeholder="e.g., 215"
            value={ this.state.weight }
            onChange={ (evt) => {
              this.setState({ weight: evt.target.value })
            }}
          />
        </div>
        <div className="input-wrapper">
          <input type="submit" value="Add" className="button"/>
        </div>
      </form>
    )
  }

}