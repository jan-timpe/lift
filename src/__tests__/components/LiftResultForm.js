import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LiftResultForm from '../../components/LiftResultForm'

describe(`LiftResultForm`, () => {

  // snapshot
  it(`should render`, () => {
    const mockAddLiftResult = jest.fn()
    const wrapper = shallow(<LiftResultForm addLiftResult={mockAddLiftResult} />)
    const component = wrapper
    expect(toJson(component)).toMatchSnapshot()
  })

  
  // form
  it(`should call addLiftResult on submit`, () => {
    const mockAddLiftResult = jest.fn()
    const preventDefault = jest.fn()
    const wrapper = shallow(<LiftResultForm addLiftResult={mockAddLiftResult} />)
    const component = wrapper

    component.find('form').simulate('submit', { preventDefault })
    expect(mockAddLiftResult.mock.calls.length).toEqual(1)
  })

})