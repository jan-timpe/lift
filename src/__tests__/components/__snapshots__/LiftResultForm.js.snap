// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiftResultForm should render 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="input-wrapper"
  >
    <label
      htmlFor="weight"
    >
      Add result:
    </label>
    <input
      onChange={[Function]}
      placeholder="e.g., 215"
      type="number"
      value=""
    />
  </div>
  <div
    className="input-wrapper"
  >
    <input
      className="button"
      type="submit"
      value="Add"
    />
  </div>
</form>
`;
